port/oc_log_internal.h

```diff
@@ -79,8 +80,24 @@
 #endif /* !defined(OC_DEBUG) && !defined(OC_PUSHDEBUG) */
 #endif
 
+#ifndef PRINT_LOG
+#define PRINT_LOG(...) printf(__VA_ARGS__)
+#endif /* !PRINT_LOG */
+
 #ifndef PRINT
-#define PRINT(...) printf(__VA_ARGS__)
+#define PRINT(...)                                                             \
+  do {                                                                         \
+    oc_logger_t *logger = oc_log_get_logger();                                 \
+    if (logger->fn != NULL) {                                                  \
+      if (logger->level < (OC_LOG_LEVEL_DEBUG)) {                              \
```

Comment:
Hmm, this might breaks apps binaries or even CTT tool (it might read something from stdout). For example cloud_server will now print usage only if you compile it with debug level logs enabled, which doesn't seem right to me. 

Either we need to switch everything from apps, onboarding_tool and python to PRINT_LOG. Or we remove the usage of PRINT in library code - I've tried to do this in https://github.com/iotivity/iotivity-lite/pull/432 .

Reply:
Ok. I merged you changes to my PR